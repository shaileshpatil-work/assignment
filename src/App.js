import { useState } from 'react';
import Card from './components/Card';
import HighLightText from './components/HighlightText';
import MainHeading from './components/MainHeading';
import Paragraph from './components/Paragraph';
import Popup from './components/Popup';
import PopupBtn from './components/PopupBtn';
import CardConstants from './constants/CardConstants';

const App = () => {
  const [visiblePopup, setPopup] = useState(false);
  const BUTTONS = ['POPULATION HEALTH', 'EPOCRATES'];
  const handlePopup = (state) => {
    setPopup(state)
  }

  return (
    <div className='container main-wrapper'>
      <div className='mx-auto w-75 my-4'>
        <MainHeading classes='text-center mb-1'>Our Products</MainHeading>
        <Paragraph alignment='text-center'>Our cloud-based and on-premises solutions deliver measurable financial and clinical results for healthcare organizations of all shapes and sizes. That's why our top-performing clients are beating industry becnmarks across the board.</Paragraph>
      </div>
      <div className='row custom-row'>
        <div className='col-sm-12 col-md-6'>
          <div className='left-section mb-0 mb-3'>
              <div className='banner-wrapper'>
                <h4>Electronic Heath Records</h4>
                <Paragraph color='text-white' spacing='mt-2 mb-2'>All your patient data organized and available so you can be entirely in the moment of care.</Paragraph>
                <a href='https://www.google.com/' className='text-decoration-none' target='_blank' rel='noreferrer'><HighLightText color='highligtContent'>Streamline documentation</HighLightText></a>                
              </div>            
          </div>
        </div>
        <div className='col-sm-12 col-md-6 right-section'>
          <div className='row custom-row'>
            {
              CardConstants.map((card) => {
                return(                  
                  <div className='col-sm-6' key={card.id}>
                    <Card title={card.title} subTitle={card.subTitle} highlight={card.highlight} link={card.link}/>
                  </div>                  
                )
              })
            }                        
          </div>
        </div>
      </div>
      <div className='row custom-row mt-3'>                     
          <div className='col-sm-12 col-md-6 mb-0 mb-3'>
            <PopupBtn popup={handlePopup}>POPULATION HEALTH</PopupBtn>
          </div>
        <div className='col-sm-12 col-md-6 mb-0 mb-3'>
          <PopupBtn popup={handlePopup}>EPOCRATES<sup className='trademark'>&#174;</sup></PopupBtn>
        </div>         
      </div>
      {visiblePopup && <Popup popup={handlePopup} cards={CardConstants.slice(0, 2)}/>}
    </div>
  );
}

export default App;
