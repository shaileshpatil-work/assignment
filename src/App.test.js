import { render, screen } from '@testing-library/react';
import App from './App';

describe('App component content testing', () => {

  test('App component snapshot testing', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('headingElement should have Our Products name', () => {
    render(<App/>);
    const headingElement = screen.getByText(/Our Products/i);
    expect(headingElement).toBeInTheDocument();
  });

  test('bannerHeadingElement should have Electronic Heath Records name', () => {
    render(<App />);
    const bannerHeadingElement = screen.getByText(/Electronic Health Records/i);
    expect(bannerHeadingElement).toBeInTheDocument();
  });
});
