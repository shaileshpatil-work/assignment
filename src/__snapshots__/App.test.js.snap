// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component content testing App component snapshot testing 1`] = `
<DocumentFragment>
  <div
    class="container main-wrapper"
  >
    <div
      class="mx-auto w-75 my-4"
    >
      <h1
        class="text-center mb-1"
      >
        Our Products
      </h1>
      <p
        class="text-center"
      >
        Our cloud-based and on-premises solutions deliver measurable financial and clinical results for healthcare organizations of all shapes and sizes. That's why our top-performing clients are beating industry becnmarks across the board.
      </p>
    </div>
    <div
      class="row custom-row"
    >
      <div
        class="col-sm-12 col-md-6"
      >
        <div
          class="left-section mb-0 mb-3"
        >
          <div
            class="banner-wrapper"
          >
            <h4>
              Electronic Heath Records
            </h4>
            <p
              class="mt-2 mb-2 text-white"
            >
              All your patient data organized and available so you can be entirely in the moment of care.
            </p>
            <a
              class="text-decoration-none"
              href="https://www.google.com/"
              rel="noreferrer"
              target="_blank"
            >
              <span
                class="highligtContent"
              >
                Streamline documentation
              </span>
            </a>
          </div>
        </div>
      </div>
      <div
        class="col-sm-12 col-md-6 right-section"
      >
        <div
          class="row custom-row"
        >
          <div
            class="col-sm-6"
          >
            <div
              class="card card-wrapper box-shadow h-100"
            >
              <a
                class="text-decoration-none"
                href="https://www.cybage.com/"
                rel="noreferrer"
                target="_blank"
              >
                <div
                  class="card-body text-center"
                >
                  <h3
                    class="subheading"
                  >
                    MEDICAL BILLING
                  </h3>
                  <p
                    class="mb-1"
                  >
                    Cure your billing woes with an automated claims process that measurably improves collections.
                  </p>
                  <span
                    class="highligtContent"
                  >
                    Get paid faster
                  </span>
                </div>
              </a>
            </div>
          </div>
          <div
            class="col-sm-6"
          >
            <div
              class="card card-wrapper box-shadow h-100"
            >
              <a
                class="text-decoration-none"
                href="https://getbootstrap.com/"
                rel="noreferrer"
                target="_blank"
              >
                <div
                  class="card-body text-center"
                >
                  <h3
                    class="subheading"
                  >
                    PATIENT ENGAGEMENT
                  </h3>
                  <p
                    class="mb-1"
                  >
                    Reduce no-shows and stay connected through every stage of there visit with our robust patient portal.
                  </p>
                  <span
                    class="highligtContent"
                  >
                    Stay connected
                  </span>
                </div>
              </a>
            </div>
          </div>
          <div
            class="col-sm-6"
          >
            <div
              class="card card-wrapper box-shadow h-100"
            >
              <a
                class="text-decoration-none"
                href="https://getbootstrap.com/"
                rel="noreferrer"
                target="_blank"
              >
                <div
                  class="card-body text-center"
                >
                  <h3
                    class="subheading"
                  >
                    CARE COORDINATION
                  </h3>
                  <p
                    class="mb-1"
                  >
                    Share secure data and collaborate with internal and external care teams more easily.
                  </p>
                  <span
                    class="highligtContent"
                  >
                    Coordinate seamlessly
                  </span>
                </div>
              </a>
            </div>
          </div>
          <div
            class="col-sm-6"
          >
            <div
              class="card card-wrapper box-shadow h-100"
            >
              <a
                class="text-decoration-none"
                href="https://www.google.com/"
                rel="noreferrer"
                target="_blank"
              >
                <div
                  class="card-body text-center"
                >
                  <h3
                    class="subheading"
                  >
                    MOBILE CAPABILITIES
                  </h3>
                  <p
                    class="mb-1"
                  >
                    Access medical information and take action on urgent taks anywhere, anytime from your mobile device.
                  </p>
                  <span
                    class="highligtContent"
                  >
                    Pratice On-the-go
                  </span>
                </div>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="row custom-row mt-3"
    >
      <div
        class="col-sm-12 col-md-6 mb-0 mb-3"
      >
        <a
          class="d-flex justify-content-center p-4 bg-white text-decoration-none box-shadow subheading"
          role="button"
        >
          POPULATION HEALTH
        </a>
      </div>
      <div
        class="col-sm-12 col-md-6 mb-0 mb-3"
      >
        <a
          class="d-flex justify-content-center p-4 bg-white text-decoration-none box-shadow subheading"
          role="button"
        >
          EPOCRATES
          <sup
            class="trademark"
          >
            Â®
          </sup>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
